.github/workflows/render.yml

```diff
@@ -36,11 +37,42 @@ jobs:
           name: newsletters
           path: out/html
 
-      - name: Deploy 🚀
-        uses: JamesIves/github-pages-deploy-action@3.7.1
+  deploy:
+    name: Deploy to GitHub Pages
+    runs-on: ubuntu-latest
+    needs: build
+    if: ${{ github.ref == 'refs/heads/main' }}
+    # Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+    # However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+    concurrency:
+      group: "pages"
+      cancel-in-progress: false
+    # Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+    permissions:
+      contents: read
+      pages: write
+      id-token: write
+    environment:
+      name: github-pages
+      url: ${{ steps.deploy.outputs.page_url }}
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - uses: actions/download-artifact@v3
```

Comment:
Issue: Here, we download the artifact just to upload it as a different name.

Suggestion: It might be better to move the `actions/upload-pages-artifact` to the `build` job, and just call `deploy-pages` here. That way, there's no duplicate artifact, but we can still download the raw files should we want to.

Reply:
Only having an auth token with permissions when we are deploying static files rather than running potentially arbitrary code seems nice